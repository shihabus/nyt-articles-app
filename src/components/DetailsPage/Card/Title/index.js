import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
    container: {
        paddingVertical: 5,
        borderBottomColor:'#fff',
        borderBottomWidth:0.3
    },
    title: {
        fontSize: 25,
        color: '#000',
        fontWeight: 'bold',
        lineHeight: 30,
        textAlign:'justify'
    },
    rowContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingVertical: 10
    },
    author: {
        color: '#000',
    },
})

export const TitleSection = ({ title, author, date }) => {
    return (
        <View style={styles.container}>
            <Text style={styles.title}>{title}</Text>
            <View style={styles.rowContainer}>
                <Text style={styles.author}>{author === null ? '--' : author}</Text>
                <Text>{` ${date}`}</Text>
            </View>
        </View>
    )
}

TitleSection.propTypes={
    title:PropTypes.string, 
    author:PropTypes.string, 
    date:PropTypes.string
}

