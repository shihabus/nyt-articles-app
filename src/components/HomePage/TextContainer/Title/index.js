import React from "react";
import { Text,StyleSheet } from "react-native";
import PropTypes from "prop-types";

const styles = StyleSheet.create({
  titleText: {
    textAlign: "justify",
    fontSize: 16
  }
});

const Title = ({ title }) => {
  return (
    <Text numberOfLines={2} ellipsizeMode="tail" style={styles.titleText} testID='Title'>
      {title}
    </Text>
  );
};

Title.propTypes = {
  title: PropTypes.string
};

export default Title;
