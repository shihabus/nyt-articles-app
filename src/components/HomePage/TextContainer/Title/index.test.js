/* eslint-disable no-undef */
// /* eslint-disable import/named */
// /* eslint-disable no-undef */
// /**
//  * @jest-environment jsdom
//  */

import React from "react";
import { shallow } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import Title from "./index";
import { checkProps,findComponent } from "../../../../../testUtils";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

const setUp = (props = {}) => {
  const wrapper = shallow(<Title {...props} />);
  return wrapper;
};

describe("Title", () => {
  let wrapper;
  beforeEach(() => {
    const props = {
        title: 'Test'
    };
    wrapper = setUp(props);
  });

  it("Snapshot", () => {
    expect(wrapper).toMatchSnapshot("Title");
  });

  it("Checked props", () => {
    const expectedProps = {
        title: 'Test'
    };
    const propsErr = checkProps(Title, expectedProps);
    expect(propsErr).toBeUndefined();
  });

  it('should render child',()=>{
    expect(findComponent(wrapper,"Title").length).toEqual(1)
  })
});
