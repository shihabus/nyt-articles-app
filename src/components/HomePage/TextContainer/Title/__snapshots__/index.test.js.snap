// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Snapshot: Title 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={2}
  style={
    Object {
      "fontSize": 16,
      "textAlign": "justify",
    }
  }
  testID="Title"
>
  Test
</Text>
`;
